File: SWEA/D3/1213. ［S／W 문제해결 기본］ 3일차 － String/［S／W 문제해결 기본］ 3일차 － String.py
for test_case in range(1, 11):
    t = int(input())
    find = input()
    word = input()
    cnt = 0
    for i in range(len(word)-len(find)+1):
        result = ""
        num = len(find)
        result += "".join(word[i:i+num])

        if result == find:
            cnt += 1
    print(f'#{test_case}', cnt)

File: SWEA/D3/1213. ［S／W 문제해결 기본］ 3일차 － String/［S／W 문제해결 기본］ 3일차 － String.java
import java.util.*;
import java.io.*;

class Solution {
  public static void main(String[] args) throws Exception {
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    StringTokenizer st;
    int T = 10;

    for(int test_case=1; test_case<=T; test_case++) {
      int tc = Integer.parseInt(br.readLine());
      String find = br.readLine();
      String str = br.readLine();

      int findLen = find.length();

      int cnt = 0;
      for(int i=0; i<str.length()-findLen+1; i++) {
        String word = str.substring(i, i+findLen);
        if(word.equals(find)) {
          cnt++;
        }
      }
      System.out.println("#" + test_case + " " + cnt);
    }
  }
}